### PR TITLE
Mes 1249 Validation of waiting room fields

### PR DESCRIPTION
**Pull request by Jason-Pullen**
mes-1249 proceed to vehicle check

e2e tests pending

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [] branch rebased against the latest develop
- [] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [x] Tested by QA
- [x] PO's approval

Screenshot 2019-03-25 at 11 54 59
Screenshot 2019-03-25 at 11 55 15
Screenshot 2019-03-25 at 11 55 34

### CODE REVIEW

File: src/theme/sass-partials/_validation.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
+.validation-bar{
+    border-color: transparent;
+    border-left-style: solid;
+    border-left-width: 8px;
+    &.ng-invalid {
+      border-color: map-get($colors, mes-validation-error)10e1e;;
```
Comment (ammarhaiderbjss) [inline src/theme/sass-partials/_validation.scss]: typo?

Comment (Jason-Pullen) [reply src/theme/sass-partials/_validation.scss]: defo.. corrected

**Comment by johnmorleybjss** [comment]
One thing I notice on this page is that it scrolls and bounces. Would be good to make it all fit on the page without scrolling like in the design + add `no-bounce` to the `ion-content`

File: src/pages/waiting-room/waiting-room.html [hunk 2]
```diff
@@ -15,49 +15,116 @@
...
+                which
+                satisfies the requirements of the legislation.
+              </p>
+            </li>
+          </ul>
+        </ion-col>
+      </ion-row>
+      <ion-row>
+        <div class="validation-bar"
+          [class.ng-invalid]="!form.value['insuranceCheckboxCtrl'] && form.get('insuranceCheckboxCtrl').dirty"></div>
```
Comment (rstepnowski) [inline src/pages/waiting-room/waiting-room.html]: Can we extract this logic into a separate function and parametrise it to remove code duplication?
eg.:
`function isFormFieldDirty(XYZ) { 
return !form.value[XYZ] && form.get(XYZ).dirty
}`

Comment (Jason-Pullen) [reply src/pages/waiting-room/waiting-room.html]: done

**Pull request merged**